motr/io_req.c

```diff
@@ -1132,6 +1133,110 @@ static int application_data_copy(struct pargrp_iomap      *map,
 	return M0_RC(0);
 }
 
+/* This function calculates and verify checksum for data read.
+ * It divides the data in multiple units and call the client api
+ * to verify checksum for each data unit.
+ */
+static bool verify_checksum(struct m0_op_io *ioo)
+{
+	struct m0_pi_seed seed;
+	struct m0_bufvec user_data = {};
+	int usz, rc, count, i;
+	struct m0_generic_pi *pi_ondisk;
+	struct m0_bufvec_cursor   datacur;
+	struct m0_bufvec_cursor   tmp_datacur;
+	struct m0_ivec_cursor     extcur;
+	uint32_t nr_seg;
+	int attr_idx = 0;
+	m0_bcount_t bytes;
+	
+	usz = m0_obj_layout_id_to_unit_size(
+			m0__obj_lid(ioo->ioo_obj));
+
+	m0_bufvec_cursor_init(&datacur, &ioo->ioo_data);
+	m0_bufvec_cursor_init(&tmp_datacur, &ioo->ioo_data);
+	m0_ivec_cursor_init(&extcur, &ioo->ioo_ext);
+
+	while ( !m0_bufvec_cursor_move(&datacur, 0) &&
+		!m0_ivec_cursor_move(&extcur, 0) &&
+		attr_idx < ioo->ioo_attr.ov_vec.v_nr){
+
+		/* calculate number of segments required for 1 data unit */
+		nr_seg = 0;
+		count = usz;
+		while (count > 0) {
+			nr_seg++;
+			bytes = m0_bufvec_cursor_step(&tmp_datacur);
+			if (bytes < count) {
+				m0_bufvec_cursor_move(&tmp_datacur, bytes);
+				count -= bytes;
+			}
+			else {
```

Comment:
Please fix the styling of if-else blocks, here and elsewhere.

if() {
...
} else {
...
}